### PR TITLE
CI: add regen job

### PR DESCRIPTION
**Pull request by fredyshox**
Simple test using regen_all on 2 segments: TOYOTA (old EON) and FORD (recent TICI). Only thing tested here is if openpilot was engaged for min. of 10 seconds.

### CODE REVIEW

**Comment by adeebshihadeh** [comment]
These segments aren't that unique. I was thinking this is a separate segment list that would have interesting cases like comma two, pandaState, etc.

**Comment by fredyshox** [comment]
prius segments its kinda unique on its own, and covers a lot of cases: comma 2, no peripheral state, old panda state

**Comment by haraschax** [comment]
This test is still fairly slow and only covers 2 segments. Is there any downside to only covering 5-10% of segment?

**Comment by fredyshox** [comment]
I feel the main goal of this test is to see if full op replay still works - so even the single segment would be fine.
Tests for migration mechanism for old files and edge cases could run individual processes in isolation (like controlsd).

File: selfdrive/test/process_replay/test_regen.py [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+
+from parameterized import parameterized
+
+from openpilot.selfdrive.test.process_replay.regen import regen_segment
+from openpilot.selfdrive.test.process_replay.process_replay import check_openpilot_enabled, CONFIGS
+from openpilot.selfdrive.test.openpilotci import get_url
+from openpilot.tools.lib.logreader import LogReader
+from openpilot.tools.lib.framereader import FrameReader
+
+EXCLUDED_PROCESSES = {"dmonitoringd", "dmonitoringmodeld"}
```
Comment (adeebshihadeh) [inline selfdrive/test/process_replay/test_regen.py:13]: why is DM disabled? does the extra framereader significantly slow things down?

File: .github/workflows/selfdrive_tests.yaml [hunk 2]
```diff
@@ -228,6 +228,34 @@ jobs:
...
+      run: eval "$BUILD"
+    - name: Build Docker image
+      run: eval "$BUILD_CL"
+    - name: Build openpilot
+      run: |
+        ${{ env.RUN }} "scons -j$(nproc)"
+    - name: Run regen
+      timeout-minutes: 30
+      run: |
+        ${{ env.RUN_CL }} "unset PYTHONWARNINGS && \
```
Comment (adeebshihadeh) [inline .github/workflows/selfdrive_tests.yaml:255]: can you leave a comment about why PYTHONWARNINGS has to be unset?

File: selfdrive/test/process_replay/test_regen.py [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+  @parameterized.expand(TESTED_SEGMENTS)
+  def test_engaged(self, case_name, segment):
+    tested_procs = [p for p in CONFIGS if p.proc_name not in EXCLUDED_PROCESSES]
+
+    route, sidx = segment.rsplit("--", 1)
+    lr, frs = ci_setup_data_readers(route, sidx)
+    output_logs = regen_segment(lr, frs, processes=tested_procs, disable_tqdm=True)
+
+    engaged = check_openpilot_enabled(output_logs)
+    self.assertTrue(engaged, f"Openpilot not engaged in {case_name}")
```
Comment (adeebshihadeh) [inline selfdrive/test/process_replay/test_regen.py:44]: openpilot is stylized as always lowercase
```suggestion
    self.assertTrue(engaged, f"openpilot not engaged in {case_name}")
```

File: selfdrive/test/process_replay/test_regen.py (hunk 1 above)
Comment (fredyshox) [reply selfdrive/test/process_replay/test_regen.py:13]: The segments used are public and are missing dcamera.hevc

File: .github/workflows/selfdrive_tests.yaml (hunk 2 above)
Comment (fredyshox) [reply .github/workflows/selfdrive_tests.yaml]: actually it works with PYTHONWARNINGS set. its possible that we can remove it from all model related tests
